### ISSUE TITLE
Add support for handling multiple spaces between words in WindowedWords iterator

### ISSUE THREAD

**Issue by andrewgazelka** [Eventual-Inc/Daft#3133]
### Description
Currently, the `WindowedWords` iterator doesn't properly handle text with multiple consecutive spaces between words. The functionality is intentionally disabled (commented out) in the test suite for performance reasons.

#### Current Behavior
- The implementation uses `memchr::memchr_iter(b' ', text.as_bytes())` to find space characters
- Each space is treated as a word boundary without considering consecutive spaces
- Extra spaces are included in the returned slices, which can lead to inconsistent results

#### Desired Behavior
As shown in the commented test case:
```rust
#[test]
fn test_multiple_spaces_between_words() {
    let s = "Hello    world  from  Rust";
    let result: Vec<&str> = s.windowed_words(2).collect();
    assert_eq!(result, vec!["Hello    world", "world  from", "from  Rust"]);
}
```

The iterator should:
1. Preserve all spaces between words in the output slices
2. Treat consecutive spaces as a single word boundary
3. Maintain the exact spacing in the output windows

**Comment by jaychia** [comment]
Can we have more context in this PR from a user-perspective? I see that this has to do with the minhash kernel.

- Is this a regression of functionality?
- Does it affect correctness of our minhash kernel?
- Is this urgent?

**Comment by andrewgazelka** [comment]
I don't believe it's a regression in functionality. Originally, the behavior was like this. It doesn't affect correctness per se, as it's still correct, but it could affect expected behavior since people might expect multiple spaces to be treated as one. I don't think this is necessarily urgent unless someone has an issue with it.
